submissions/nadsatt/friends-app/modules/filter-items.js

```diff
@@ -0,0 +1,173 @@
+import svgs from './svgs.js';
+
+class Filter {
+    constructor(optionsHTML, property, category, userService){
+        this.element = document.createElement('li');
+        this.element.innerHTML =
+            `<div class="filter-item__options">
+                ${optionsHTML}
+                <span class="filter-item__option" data-state="none">${svgs.none}</span>
+            </div>`;
+        this.element.classList.add('filter-item', `${category}-filter-item`);
+
+        this.property = property;
+        this.category = category;
+        this.name = `${property}-${category}`;
+        this.userService = userService;
+        this.defaultState = 'none';
+        this.defaultOptionElement = this.element.querySelector(`[data-state="${this.defaultState}"]`);
+        this.settedState = this.defaultState;
+        this.activatedOptionElement = this.defaultOptionElement;
+        this.classNames = {
+            openedFilterElement: 'filter-item--opened',
+            activatedOptionElement: 'filter-item__option--activated'
+        };
+    }
+
+    open(){
+        this.element.classList.add(this.classNames.openedFilterElement);
+    }
+
+    close(){
+        this.element.classList.remove(this.classNames.openedFilterElement);
+    }
+
+    setState(optionElement){
+        this.deactivateOptionAndIcon();
+        this.updateOptionAndState(optionElement);
+        if(this.settedState !== this.defaultState) this.activateOptionAndIcon();
+    }
+
+    applyAccordingToState(){
+        if(this.settedState !== this.defaultState){
+            this[`applyAccordingTo${this.getCapitalizedState(this.settedState)}State`]();
```

Comment:
Bad idea :) Please, write less generic and verbose code here.

``applyAccordingTo${this.getCapitalizedState(this.settedState)}State`` not understandable to other developers :)

Reply:
```javascript
class InputStatesKeeper {
    // some state
    // usersService
    // this.setState
    // ...
}

class Sorter extends InputStatesKeeper {
    constructor(ascSvg, descSvg, ...args){
        const optionsHTML =
            `<span class="filter-item__option" data-state="asc">${ascSvg}</span>
             <span class="filter-item__option" data-state="desc">${descSvg}</span>`;
        super(optionsHTML, ...args);
    }

    handleAscending(){}
    handleDescending(){}

    resetState() {
        this.setState(this.defaultOptionElement);
    }
}

class NumberSortFilter extends Sorter {
    constructor(...args){
        super(svgs.ascNumber, svgs.descNumber, ...args);

        this.ascComparator = function(a, b){
            return a[this.property] - b[this.property];
        };

        this.descComparator = function(a, b){
            return b[this.property] - a[this.property];
        };
    }

    handleAscending(){
        const users = this.usersService.getUsers();
        users.sort(this.ascComparator)
        this.usersService.updateUsers(users)
    }
}
```